src/test/kotlin/com/sourcegraph/cody/agent/protocol/util/Rfc3986UriEncoderTest.kt

```diff
@@ -0,0 +1,16 @@
+package com.sourcegraph.cody.agent.protocol.util
+
+import junit.framework.TestCase
+
+class Rfc3986UriEncoderTest : TestCase() {
+
+  fun `test encode Windows path`() {
+    val fixedUri = Rfc3986UriEncoder.encode("file:///C:/Users/user/Test.java")
```

Comment:
Can you please add test for `"file:///c:/Users/user/Test.java"`?

Reply:
Sure!

Reply:
BTW can you elaborate @pkukielka? It seems to me that partitions are always capitalized.

Reply:
Well, maybe it's defensive programming, but windows theoretically is case insensitive, so what guarantees you that you won't get lower letter there somehow?
I understand that we do not expect it to happen :)

Reply:
Fixed